src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -683,7 +731,10 @@ private void OnDynamicFileInfoUpdated(object sender, string dynamicFilePath)
                 return;
             }
 
-            _sourceFiles.ProcessFileChange(dynamicFilePath, fileInfoPath);
+            if (fileInfoPath != null)
+            {
+                _sourceFiles.ProcessFileChange(dynamicFilePath, fileInfoPath);
+            }
```

Comment:
why is `if (!_dynamicFilePathMaps.TryGetValue(dynamicFilePath, out var fileInfoPath))` not under lock?  have you audited all usages of _dynamicFilePathMaps?

Reply:
Re-audited, thanks for looking.